examples/vision/attention_mll_classification.py

```diff
@@ -0,0 +1,579 @@
+label for a set of images.
+
+The motivation behind this model is to:
+* know which instance (in this case: image) has contributed the most to the class label
+prediction.
+* leverage from weakly labelled samples.
+
+### MIL Classifier: Attention-based approach
+
+The MIL classifier is modelled using neural networks. The attention mechanism as MIL pooling:
+* Trainable MIL operator.
+* Interpretable results using attention scores.
+
+Using attention scores or weights could be very useful for results interpretability as it will
+not merely predict the final output class but the instance scores that led to those
+classification results.
+
+Another reason why this method could be helpful is: weakly labelled data. Since labelling
+individual samples is often perfomed manually, it is considered to be a time consuming task.
+This task is also prone to errors (especially in computer vision problems).
+
+Therefore, instead of having individual labelled data points or instances, this algorithm
+receives a single label for each bag of instances. The bag is labelled positive if it contains
+atleast one positive instance and negative if it does not contain any.
+
+The attention mechanism provides insight into the contribution of each instance to the bag label.
+
+Besides the implementation of this mechanism, some **regularization** techniques,
+**ensemble averaging** (for stability) and dealing with **imbalanced data** will be covered.
+
+## References
+
+- The paper [Attention-based Deep Multiple Instance Learning](https://arxiv.org/pdf/1802.04712.pdf)
+is authored by Maximilian Ilse, Jakub M. Tomczak and Max Welling.
+- Some of attention operator code implementation was inspired from https://github.com/utayao/Atten_Deep_MIL.
+- Imbalanced data [tutorial](https://www.tensorflow.org/tutorials/structured_data/imbalanced_data)
+by TensorFlow.
+"""
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import datasets
+from tensorflow.keras import layers
+from tensorflow.keras import callbacks
+import os
+import tempfile
+from tqdm import tqdm
+from matplotlib import pyplot as plt
+
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+plt.style.use("ggplot")
+
+"""
+## Create Dataset
+
+In this example, either of `mnist` or `fashion_mnist` data can be used.
+At first we will set up the configurations and then prepare the datasets.
+
+### Configurations
+
+- `DATASET`: Either mnist digits or fashion mnist data.
+- `POS_CLASS`: The desired class to be kept in the positive bag.
+- `BAG_COUNT`: The number of training bags.
+- `VAL_BAG_COUNT`: The number of validation bags.
+- `BAG_SIZE`: The number of instances in a bag.
+- `PLOT_SIZE`: The number of bags to plot.
+- `ENSEMBLE_AVG_COUNT`: The number of models to create and get their average. (OPTIONAL:
+often results in better performance - set to 1 for single model)
+"""
+
+DATASET = {"mnist": datasets.mnist, "fashion_mnist": datasets.fashion_mnist}
```

Comment:
You can just hard-code one of them since only one is used in the example. Users can always modify the example themselves

Reply:
Sure, I will just provide one then.